tt_metal/impl/dispatch/kernels/command_queue_producer.hpp

```diff
@@ -304,9 +305,15 @@ void produce_for_eth_src_router(
         const uint32_t src_buf_type = command_ptr[4];
         const uint32_t src_page_index = command_ptr[6];
 
-        uint32_t fraction_of_producer_cb_num_pages = consumer_cb_num_pages / 2;
+        bool read_from_sysmem = (BufferType)src_buf_type == BufferType::SYSTEM_MEMORY;
+        if (!read_from_sysmem) {
+            // Data needs to come from target remote device
+            continue;
```

Comment:
this is added to prevent issue queue reader from sending garbage data from its local device to the src router